modules/src/core/ics02_client/consensus_state.rs

```diff
@@ -0,0 +1,92 @@
+use crate::prelude::*;
+
+use core::marker::{Send, Sync};
+
+use dyn_clone::DynClone;
+use erased_serde::Serialize as ErasedSerialize;
+use ibc_proto::google::protobuf::Any;
+use ibc_proto::protobuf::Protobuf as ErasedProtobuf;
+
+use crate::core::ics02_client::client_type::ClientType;
+use crate::core::ics02_client::error::Error;
+use crate::core::ics23_commitment::commitment::CommitmentRoot;
+use crate::dynamic_typing::AsAny;
+use crate::timestamp::Timestamp;
+
+/// Abstract of consensus state information used by the validity predicate
+/// to verify new commits & state roots.
+///
+/// Users are not expected to implement sealed::ErasedPartialEqConsensusState.
+/// Effectively, that trait bound mandates implementors to derive PartialEq,
+/// after which our blanket implementation will implement
+/// `ErasedPartialEqConsensusState` for their type.
+pub trait ConsensusState:
+    AsAny
+    + sealed::ErasedPartialEqConsensusState
+    + DynClone
+    + ErasedSerialize
+    + ErasedProtobuf<Any, Error = Error>
+    + core::fmt::Debug
+    + Send
```

Comment:
Just a note: Penumbra and Anoma use [tower-abci](https://github.com/penumbra-zone/tower-abci), an async implementation of ABCI. I was worried that this `Send + Sync` might be an issue, but it seems like they're required extensively in that library anyways, so we're probably OK.